contracts/contracts/lib/LibGateway.sol

```diff
@@ -405,11 +447,24 @@ library LibGateway {
         // should increase the appliedNonce to allow the execution of the next message
         // of the batch (this is way we have this after the nonce logic).
         if (!crossMsg.to.subnetId.equals(s.networkName)) {
+            CrossMessageValidationOutcome outcome = validateCrossMessage(crossMsg);
+            if (outcome != CrossMessageValidationOutcome.Valid) {
+                sendReceipt(
+                    crossMsg,
+                    OutcomeType.SystemErr,
+                    abi.encodeWithSelector(InvalidXnetMessage.selector, validationOutcomeToInvalidXnetMsgReason(outcome))
+                );
+                return;
+            }
+
             bytes32 cid = crossMsg.toHash();
+            s.postboxKeys.add(cid);
```

Comment:
if the cross network message is topdown, there is no point inserting into postbox? Why not just emit a topdown event or call commit topdown directly?